lib/sidekiq/cloud_watch_metrics/publisher.rb

```diff
@@ -0,0 +1,100 @@
+# frozen_string_literal: true
+
+require "sidekiq"
+
+require "aws-sdk-cloudwatch"
+
+require "sidekiq/cloud_watch_metrics/collector"
+
+module Sidekiq::CloudWatchMetrics
+  class Publisher
+    begin
+      require "sidekiq/util"
+      include Sidekiq::Util
+    rescue LoadError
+      # Sidekiq 6.5 refactored to use Sidekiq::Component
+      require "sidekiq/component"
+      include Sidekiq::Component
+    end
+
+    INTERVAL = 60 # seconds
+
+    def initialize(config: Sidekiq,
+                   client: Aws::CloudWatch::Client.new,
+                   namespace: "Sidekiq",
+                   collector: nil,
+                   **collector_kwargs)
+      # Sidekiq 6.5+ requires @config, which defaults to the top-level
+      # `Sidekiq` module, but can be overridden when running multiple Sidekiqs.
+      @config = config
+      @client = client
+      @namespace = namespace
+
+      if collector_kwargs.size > 0
+        warn <<~TEXT
```

Comment:
I suggest to remove the support for `collector_kwargs` in the future and complete the deprecation in 2.7.